assets/js/payment-methods-demo/index.js

```diff
@@ -10,9 +10,23 @@ import {
  * Internal dependencies
  */
 import { ExpressApplePay, ExpressPaypal } from './express-payment';
-import { PaypalPaymentMethod, ccPaymentMethod } from './payment-methods';
+import { paypalPaymentMethod, ccPaymentMethod } from './payment-methods';
 
-registerExpressPaymentMethod( 'applepay', ExpressApplePay );
-registerExpressPaymentMethod( 'paypal', ExpressPaypal );
-registerPaymentMethod( 'paypal', PaypalPaymentMethod );
-registerPaymentMethod( 'cc', ccPaymentMethod );
+registerExpressPaymentMethod(
+	( Config ) =>
+		new Config( {
+			id: 'applepay',
+			activeContent: ExpressApplePay,
```

Comment:
This presents an interesting convention question: should a property which contains a React component be capitalized? That is, should this be `ActiveContent`? It is copied into a capitalized variable again (`ExpressPaymentMethod`) before being used, but should it also be capitalized here? I can see arguments on both sides but I'm curious what you think.

A related, but separate question: should this be a React element/node, rather than a component? That is, should it be `activeContent: <ExpressPaypal />`, which matches the use of `activeContent` in composite-checkout? This would mean that if you wanted to pass additional props to the component, you'd need to clone it before rendering, but that is relatively inexpensive to do and it would allow specifying props at register time as well, which could possibly be valuable (as one random example: reusing the same component for different contents). (The other reason we did this in composite-checkout is that passing an element as a prop is a much more common pattern than passing a component, although I realize that you don't really use these as props anywhere).

Reply:
> should a property which contains a React component be capitalized? That is, should this be ActiveContent? It is copied into a capitalized variable again (ExpressPaymentMethod) before being used, but should it also be capitalized here? 

Hmm, I've actually argued _for capitalizing_ before and I think it's a good signal.  I just missed thinking about that when I wrote this.  So I think I'll do that.

> A related, but separate question: should this be a React element/node, rather than a component? 

Ooo... those are good points.  I saw that `composite-checkout` was using React nodes and couldn't think of why that'd be necessary, but I definitely see how it could be useful.  As a bonus it also makes it easier to validate using `React.isValidElement`.  Thanks for the suggestions -I'm going to implement.